packages/flutter_tools/lib/src/isolated/native_assets/native_assets.dart

```diff
@@ -410,417 +565,363 @@ Uri nativeAssetsBuildUri(Uri projectUri, OS os) {
-        );
-      } else if (const LocalPlatform().isWindows) {
-        nativeAssetsYaml = await dryRunNativeAssetsWindows(
-          projectUri: projectUri,
-          flutterTester: true,
-          fileSystem: fileSystem,
-          buildRunner: buildRunner,
+    case OS.macOS:
+      if (flutterTester) {
+        await copyNativeCodeAssetsMacOSFlutterTester(
+          buildUri,
+          fatAssetTargetLocationsMacOS(codeAssets, buildUri),
+          codesignIdentity,
+          buildMode,
+          fileSystem,
         );
       } else {
-        await nativeBuildRequired(buildRunner);
-        nativeAssetsYaml = null;
+        await copyNativeCodeAssetsMacOS(
+          buildUri,
+          fatAssetTargetLocationsMacOS(codeAssets, null),
+          codesignIdentity,
+          buildMode,
+          fileSystem,
+        );
       }
-    case build_info.TargetPlatform.linux_arm64:
-    case build_info.TargetPlatform.linux_x64:
-      nativeAssetsYaml = await dryRunNativeAssetsLinux(
-        projectUri: projectUri,
-        fileSystem: fileSystem,
-        buildRunner: buildRunner,
-      );
-    case build_info.TargetPlatform.windows_arm64:
-    case build_info.TargetPlatform.windows_x64:
-      nativeAssetsYaml = await dryRunNativeAssetsWindows(
-        projectUri: projectUri,
-        fileSystem: fileSystem,
-        buildRunner: buildRunner,
-      );
-    case build_info.TargetPlatform.android_arm:
-    case build_info.TargetPlatform.android_arm64:
-    case build_info.TargetPlatform.android_x64:
-    case build_info.TargetPlatform.android_x86:
-    case build_info.TargetPlatform.android:
-      nativeAssetsYaml = await dryRunNativeAssetsAndroid(
-        projectUri: projectUri,
-        fileSystem: fileSystem,
-        buildRunner: buildRunner,
-      );
-    case build_info.TargetPlatform.fuchsia_arm64:
-    case build_info.TargetPlatform.fuchsia_x64:
-    case build_info.TargetPlatform.web_javascript:
-      await ensureNoNativeAssetsOrOsIsSupported(
-        projectUri,
-        targetPlatform.toString(),
+    case OS.iOS:
+      await copyNativeCodeAssetsIOS(
+        buildUri,
+        fatAssetTargetLocationsIOS(codeAssets),
+        codesignIdentity,
+        buildMode,
         fileSystem,
-        buildRunner,
       );
-      nativeAssetsYaml = null;
+    case OS.android:
+      await copyNativeCodeAssetsAndroid(
+          buildUri, assetTargetLocations, fileSystem);
+    default:
+      throw StateError('This should be unreachable.');
   }
-  return nativeAssetsYaml;
 }
 
-/// Dry run the native builds for multiple OSes.
-///
-/// Needed for `flutter run -d all`.
-Future<Uri?> dryRunNativeAssetsMultipleOSes({
-  required NativeAssetsBuildRunner buildRunner,
-  required Uri projectUri,
-  required FileSystem fileSystem,
-  required Iterable<build_info.TargetPlatform> targetPlatforms,
-}) async {
-  if (await nativeBuildRequired(buildRunner)) {
-    return null;
+Future<void> _copyNativeCodeAssetsToBundleOnWindowsLinux(
```

Comment:
Ditto, avoid using OSspecific names in this file.

Ditto, shows up as diff even though didn't change.

Reply:
It makes consistent naming with all the other `_copy*` methods.

I've tried to mimize the diff now as much as I could.